api/subscriptions/message_cache.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2024 The VeChainThor developers
+//
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package subscriptions
+
+import (
+	"fmt"
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/vechain/thor/v2/thor"
+)
+
+type messageCache struct {
+	cache *lru.Cache
+	mu    sync.RWMutex
+}
+
+func newMessageCache(cacheSize uint32) *messageCache {
+	if cacheSize > 1000 {
+		cacheSize = 1000
+	}
+	if cacheSize == 0 {
+		cacheSize = 1
+	}
+	cache, err := lru.New(int(cacheSize))
+	if err != nil {
```

Comment:
TODO: Can we handle this more gracefully? Currently `subscriptions.New` nor `api.New` don't return errors, so this would mess up the flow a little bit